Application/EdFi.Ods.AdminApi/Features/OdsInstanceDerivative/AddOdsInstanceDerivative.cs

```diff
@@ -0,0 +1,104 @@
+    {
+        AdminApiEndpointBuilder
+           .MapPost(endpoints, "/odsInstancesDerivatives", Handle)
+           .WithDefaultDescription()
+           .WithRouteOptions(b => b.WithResponseCode(201))
+           .BuildForVersions(AdminApiVersions.V2);
+    }
+
+    public async Task<IResult> Handle(Validator validator, IAddOdsInstanceDerivativeCommand addOdsInstanceDerivativeCommand, IMapper mapper, AddOdsInstanceDerivativeRequest request)
+    {
+        await validator.GuardAsync(request);
+        var addedOdsInstanceDerivative = addOdsInstanceDerivativeCommand.Execute(request);
+        return Results.Created($"/odsInstancesDerivatives/{addedOdsInstanceDerivative.OdsInstanceDerivativeId}", null);
+    }
+
+  
+    [SwaggerSchema(Title = "AddOdsInstanceDerivativeRequest")]
+    public class AddOdsInstanceDerivativeRequest : IAddOdsInstanceDerivativeModel
+    {
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeOdsInstanceIdDescription, Nullable = false)]
+        public int OdsInstanceId { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeDerivativeTypeDescription, Nullable = false)]
+        public string? DerivativeType { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeConnectionStringDescription, Nullable = false)]
+        public string? ConnectionString { get; set; }
+    }
+
+    public class Validator : AbstractValidator<AddOdsInstanceDerivativeRequest>
+    {
+        private readonly IGetOdsInstanceQuery _getOdsInstanceQuery;
+        private readonly string _databaseEngine;
+        public Validator(IGetOdsInstanceQuery getOdsInstanceQuery, IOptions<AppSettings> options)
+        {
+            _getOdsInstanceQuery = getOdsInstanceQuery;
+            _databaseEngine = options.Value.DatabaseEngine ?? throw new NotFoundException<string>("AppSettings", "DatabaseEngine");
+
+            RuleFor(m => m.DerivativeType).NotEmpty();
+
+            RuleFor(m => m.DerivativeType)
+                .Matches("^(ReadReplica|Snapshot)$")
+                .WithMessage(FeatureConstants.OdsInstanceDerivativeDerivativeTypeNotValid)
+                .When(m => !string.IsNullOrEmpty(m.DerivativeType));
+
+            RuleFor(m => m.OdsInstanceId)
+                .NotEqual(0)
+                .WithMessage(FeatureConstants.OdsInstanceIdValidationMessage);
+
+            RuleFor(m => m.OdsInstanceId)
+                .Must(BeAnExistingOdsInstance)
+                .When(m => !m.OdsInstanceId.Equals(0));
+
+            RuleFor(m => m.ConnectionString)
+                .NotEmpty();
+
+            RuleFor(m => m.ConnectionString)
+                .Must(BeAValidConnectionString)
+                .WithMessage(FeatureConstants.OdsInstanceConnectionStringInvalid)
+                .When(m => !string.IsNullOrEmpty(m.ConnectionString));
+
+        }
+
+        private bool BeAnExistingOdsInstance(int id)
+        {
+            try
```

Comment:
@lchavarriaga
This "try-catch" code block is no longer necessary since the "_getOdsInstanceQuery.Execute(id)" is throwing the exception when the odsInstance is not found